src/main/scala/beam/analysis/RideHailFleetAnalysis.scala

```diff
@@ -0,0 +1,393 @@
+          val vehicle = pathTraversalEvent.vehicleId.toString
+          val rideHail = vehicle.contains("rideHail")
+          val ev = pathTraversalEvent.primaryFuelType == "Electricity"
+          if (rideHail) {
+            if (ev && isCAV) {
+              collectEvent(
+                rideHailEvCav,
+                pathTraversalEvent.copy(time = pathTraversalEvent.departureTime.toDouble - 0.5),
+                vehicle,
+                pathTraversalEvent.time
+              )
+            } else if (ev && !isCAV) {
+              collectEvent(
+                ridehailEvNonCav,
+                pathTraversalEvent.copy(time = pathTraversalEvent.departureTime.toDouble - 0.5),
+                vehicle,
+                pathTraversalEvent.time
+              )
+            } else if (!ev && isCAV) {
+              collectEvent(
+                rideHailNonEvCav,
+                pathTraversalEvent.copy(time = pathTraversalEvent.departureTime.toDouble - 0.5),
+                vehicle,
+                pathTraversalEvent.time
+              )
+            } else if (!ev && !isCAV) {
+              collectEvent(
+                rideHailNonEvNonCav,
+                pathTraversalEvent.copy(time = pathTraversalEvent.departureTime.toDouble - 0.5),
+                vehicle,
+                pathTraversalEvent.time
+              )
+            }
+          }
+        }
+
+      case parkEvent: ParkingEvent =>
+        val vehicle = parkEvent.vehicleId.toString
+
+        if (rideHailEvCav.contains(vehicle)) {
+          collectEvent(rideHailEvCav, parkEvent, vehicle, parkEvent.time)
+        } else if (ridehailEvNonCav.contains(vehicle)) {
+          collectEvent(ridehailEvNonCav, parkEvent, vehicle, parkEvent.time)
+        } else if (rideHailNonEvCav.contains(vehicle)) {
+          collectEvent(rideHailNonEvCav, parkEvent, vehicle, parkEvent.time)
+        } else if (rideHailNonEvNonCav.contains(vehicle)) {
+          collectEvent(rideHailNonEvNonCav, parkEvent, vehicle, parkEvent.time)
+        }
+      case _ =>
+    }
+  }
+
+  override def createGraph(event: IterationEndsEvent): Unit = {
+    processedHour = lastHour
+    processVehicleStates()
+  }
+
+  private def collectEvent(
+    vehicleEventTypeMap: mutable.Map[String, ArrayBuffer[Event]],
+    event: Event,
+    vehicle: String,
+    eventHour: Double
+  ): Unit = {
+    val events = vehicleEventTypeMap.getOrElse(vehicle, new ArrayBuffer[Event]())
+    events += event
+    vehicleEventTypeMap(vehicle) = events.sortBy(_.getTime)
+    val hour = (eventHour / 3600).toInt
+    if (hour > processedHour) {
+      processVehicleStates()
+      processedHour = hour
+    }
+  }
+
+  private def processVehicleStates() {
+    processEvents(rideHailEvCav, true, true, "rh-ev-cav")
+    processEvents(ridehailEvNonCav, true, false, "rh-ev-nocav")
+    processEvents(rideHailNonEvCav, true, true, "rh-noev-cav")
+    processEvents(rideHailNonEvNonCav, true, false, "rh-noev-nocav")
```

Comment:
If these metrics are disabled, will it still do all this processing to compute utilization?

Reply:
in case of 'RideHailFleetAnalysis' - no, because there is a special logic to prevent that:
https://github.com/LBNL-UCB-STI/beam/pull/2269/files#diff-5096becba45f341a18195da0c403b80eR88